Dockerfile

```
@@ -6,6 +6,7 @@ COPY ./go.mod ./go.sum ./
 RUN go mod download
 
 COPY . .
-RUN go build -o ./gateway-conformance ./cmd/gateway-conformance
+ARG VERSION=dev
+RUN go build -ldflags="-X github.com/ipfs/gateway-conformance/tooling.Version=${VERSION}" -o ./gateway-conformance ./cmd/gateway-conformance
```

What do you think about sourcing the metadata from git context? Similarly to what you already do in the Makefile. To me it feels more reliable - less moving parts.

```suggestion
RUN export TAG="$(git tag --points-at HEAD | sort -V | tail -n-1)" && \
  export COMMIT="$(git rev-parse HEAD --short)" && \
  export DIRTY_SUFFIX="$(test -n "`git status --porcelain`" && echo "-dirty" || true)" && \
  export VERSION="${TAG:-$COMMIT$DIRTY_SUFFIX}" && \
  go build -ldflags="-X github.com/ipfs/gateway-conformance/tooling.Version=${VERSION}" -o ./gateway-conformance ./cmd/gateway-conformance
```

For that to work, we'd only have to make sure the tags are fetched when we're building the docker image.

Also, one thing I noticed, we create tags slightly incorrectly. If you inspect `v0`, `v0.3` and `v0.3.0` now, you'll notice that `v0` and `v0.3` are on a different commit than `v0.3.0`. We should look into why that is - might need a fix in changelog based release action, I'll try to get to it today.

---

https://github.com/pl-strflt/changelog-driven-release/pull/9 <- I think this might be the fix for the tag mismatch. I'll test it later today.

---

I think that'd create coupling / break single responsibility principle: instead of having a single argument, the docker build would depend on many side-effects to succeed (git binary, `.git` folder is around, repo is fetch'd as expected, etc).